python/lsst/ts/mtdome/power_management/slip_ring.py

```diff
@@ -0,0 +1,246 @@
+    cooling_down_time_needed : `float`
+        The time interval [UNIX tai seconds] for which the slip ring needs to
+        cool down. This is set by the state machine.
+    max_power_drawn : `float`
+        The maximum power drawn [W] during the time where the power draw
+        exceeded the continuous limit. This is set by the state machine.
+    """
+
+    def __init__(
+        self,
+        log: logging.Logger,
+        index: int,
+        disabled: bool = False,
+    ) -> None:
+        self.log = log.getChild(type(self).__name__)
+        self.index = index
+        self.disabled = disabled
+
+        self.state = SlipRingState.BELOW_LOW_LIMIT
+
+        self.below_limit_start_time = 0.0
+        self.over_limit_start_time = 0.0
+        self.time_over_limit = 0.0
+        self.cooling_down_time_needed = 0.0
+
+        self.max_power_drawn = 0.0
+
+    async def get_cool_down_time(self) -> float:
+        """Get the cool down time for this slip ring.
+
+        A simple model is assumed to calculate the cool down time. This needs
+        to be replaced with the real cool down model.
+
+        Returns
+        -------
+        float
+            The cool down time for the slip ring.
+        """
+        if self.max_power_drawn < CONTINUOUS_SLIP_RING_POWER_CAPACITY:
+            power_over_limit_fraction = 0.0
+        else:
+            power_over_limit_fraction = (
+                self.max_power_drawn - CONTINUOUS_SLIP_RING_POWER_CAPACITY
+            ) / OVER_LIMIT_POWER_AVAILABLE
+        time_over_limit_fraction = self.time_over_limit / MAXIMUM_OVER_LOW_LIMIT_TIME
+        return (
+            MAXIMUM_COOL_DOWN_TIME
+            * power_over_limit_fraction
+            * time_over_limit_fraction
+        )
+
+    async def get_available_power(self, current_power_draw: float) -> float:
+        """Get the available power.
+
+        Determine whether the maximum power is available or none at all based
+        on the current power draw and, if the power draw has been over the
+        continuous limit, for how long. If the slip ring is disabled, the
+        returned value will be zero.
+
+        Parameters
+        ----------
+        current_power_draw : `float`
+            The amount of power currently in use [W].
+
+        Returns
+        -------
+        float
+            The available power [W].
+        """
+        if self.disabled:
+            return await self._handle_disabled()
+        match self.state:
+            case SlipRingState.BELOW_LOW_LIMIT:
+                return await self._handle_below_low_mimit(current_power_draw)
+            case SlipRingState.OVER_LOW_LIMIT:
+                return await self._handle_over_low_mimit(current_power_draw)
+            case SlipRingState.COOLING_DOWN:
+                return await self._handle_cooling_down()
+            case _:
+                raise ValueError(
+                    f"encountered unknown SlipRingState {self.state.name}."
```

Comment:
I think you risk having a second exception here if `self.state` is not of `SlipRingState` type. I know you are using mypy to check but better safe than sorry. Consider replacing `{self.state.name}` with `{self.state!r}`, this will give you almost the same behavior if this is a `SlipRingState` object and will be safe in case it is not.